plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/UserRoleMappingsXO.groovy

```diff
@@ -23,11 +29,14 @@ import org.hibernate.validator.constraints.NotEmpty
 @ToString(includePackage = false, includeNames = true)
 class UserRoleMappingsXO
 {
-  @NotEmpty
+  @NotBlank
+  @UserExists(groups = Create)
   String userId
 
-  @NotEmpty
+  @NotBlank
+  @RealmExists(groups = Create)
   String realm
 
+  @RolesExist(groups = [Create, Update])
```

Comment:
Is not necessary to set create/update groups as in this case the Default group will be used